### PR TITLE
feat(LongToTimestamp):introduce new UDF to cast a long (epoch time) t…

### PR DESCRIPTION
**Pull request by Doe-Ed**
…o timestamp (SQL timestamp)

### CODE REVIEW

**Comment by Doe-Ed** [comment]
Why not divide by 1000?

We used Instant.ofEpochMilli:
    public static Instant ofEpochMilli(long epochMilli) {
        long secs = Math.floorDiv(epochMilli, 1000);
        int mos = (int)Math.floorMod(epochMilli, 1000);
        return create(secs, mos * 1000_000);
    }

File: src/test/scala/com/yotpo/metorikku/code/steps/functions/test/UserDefinedFunctionsTests.scala [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import com.yotpo.metorikku.code.steps.functions.UserDefinedFunctions._
+
+class SerDefinedFunctionsTests extends Suites (
+  new EpochMilliToTimestampTest
+)
+
+class EpochMilliToTimestampTest extends FunSuite {
+
+  test("Given a Long representing Epoch Milli returns a Timestamp") {
+    assert(epochMilliToTimestamp(1584176754000L) === new Timestamp(1584176754000L))
```
Comment (doronporat) [inline src/test/scala/com/yotpo/metorikku/code/steps/functions/test/UserDefinedFunctionsTests.scala:15]: @Doe-Ed sorry, I'm not sure I understand why the assert it working. I thought it should be - 
assert(epochMilliToTimestamp(1584176754000L) === new Timestamp(1584176754L))
Can you explain?

**Comment by Doe-Ed** [comment]
val instant: Instant = Instant.ofEpochMilli(timestamp_epoch) --> divides the long (with flooring) by 1000
Timestamp.from(instant) --> Creates a Timestamp that has the given value * 1000
    public static Timestamp from(Instant instant) {
        try {
            Timestamp stamp = new Timestamp(instant.getEpochSecond() * MILLIS_PER_SECOND);
            stamp.nanos = instant.getNano();
            return stamp;
        } catch (ArithmeticException ex) {
            throw new IllegalArgumentException(ex);
        }
    }

So that's why it works.. But looking at it, I think I can just return a new Timestamp with the Long input

File: src/test/scala/com/yotpo/metorikku/code/steps/functions/test/UserDefinedFunctionsTests.scala (hunk 1 above)
Comment (Doe-Ed) [reply src/test/scala/com/yotpo/metorikku/code/steps/functions/test/UserDefinedFunctionsTests.scala:15]: ofEpoch floorDiv by 1000
from (Timestamp) multiplies by 1000

public static Instant ofEpochMilli(long epochMilli) {
  long secs = Math.floorDiv(epochMilli, 1000);
  int mos = (int)Math.floorMod(epochMilli, 1000);
  return create(secs, mos * 1000_000);
}

public static Timestamp from(Instant instant) {
  try {
    Timestamp stamp = new Timestamp(instant.getEpochSecond() * MILLIS_PER_SECOND);
    stamp.nanos = instant.getNano();
    return stamp;
  } catch (ArithmeticException ex) {
    throw new IllegalArgumentException(ex);
  }
}

Comment (Doe-Ed) [reply src/test/scala/com/yotpo/metorikku/code/steps/functions/test/UserDefinedFunctionsTests.scala:15]: MILLIS_PER_SECOND == 1000

**Pull request merged**